### PR TITLE
Initial version of integration test framework.

### PR DESCRIPTION
**Pull request by trgill**
Signed-off-by: Todd Gill <tgill@redhat.com>

### CODE REVIEW

**Comment by trgill** [comment]
@agrover @mulkieran review please?  

I think this is just a step towards what we want.  There is still work that needs to be done around logging and reporting etc.

**Comment by trgill** [comment]
Hold on... there is a warning due to the new way things are compiled.  Will fix.

**Comment by mulkieran** [comment]
I think that the idea of splitting stratisd into a front end (the D-Bus part) and a back end (the engine part) should go back on the table.

File: stratis-tests/src/test_consts.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+pub const OK_TO_DESTROY_DEV_ARRAY: &'static str = "ok_to_destroy_dev_array";
```
Comment (mulkieran) [inline stratis-tests/src/test_consts.rs]: I don't know why something called an array has type string.

File: stratis-tests/src/blockdev_tests.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::collections::BTreeSet;
+use std::path::Path;
+use std::str::FromStr;
+
+use uuid::Uuid;
+
+use common::*;
```
Comment (trgill) [inline stratis-tests/src/blockdev_tests.rs]: Ok

**Comment by trgill** [comment]
I agree with @mulkieran  that we should put the idea of "splitting stratisd into a front end (the D-Bus part) and a back end (the engine part) " should go back on the table.

File: stratis-tests/src/blockdev_tests.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            info!("PASS: initialize of already initialzied blockdevs failed (intended)");
+        }
+    }
+
+    Ok(())
+}
+
+pub fn test_blockdevs(blockdev_paths: &Vec<&Path>) -> TestResult<()> {
+    info!("Starting BlockDev Tests...");
+    let mut devices = BTreeSet::new();
```
Comment (mulkieran) [inline stratis-tests/src/blockdev_tests.rs]: You should be able to use ```resolve_devices``` in place of this block of code here.

File: stratis-tests/src/blockdev_tests.rs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+use devicemapper::Device;
+
+use stratisd::engine::strat_engine::blockdev::BlockDev;
+use stratisd::engine::strat_engine::consts::MIN_MDA_SIZE;
+
+pub fn test_blockdev_force_flag(blockdev_paths: &Vec<&Path>) -> TestResult<()> {
+
+    let mut devices = BTreeSet::new();
+    let mut devices_copy = BTreeSet::new();
+    for path in blockdev_paths {
```
Comment (mulkieran) [inline stratis-tests/src/blockdev_tests.rs]: You can use ```resolve_devices``` here.

File: stratis-tests/src/blockdev_tests.rs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+use uuid::Uuid;
+
+use common::*;
+
+use devicemapper::Device;
+
+use stratisd::engine::strat_engine::blockdev::BlockDev;
+use stratisd::engine::strat_engine::consts::MIN_MDA_SIZE;
+
```
Comment (mulkieran) [inline stratis-tests/src/blockdev_tests.rs]: A comment here, stating the property being tested would be helpful.

**Comment by trgill** [comment]
Replaced by PR #199

**Pull request closed without merging**